Detectors/GlobalTracking/include/GlobalTracking/MatchGlobalFwd.h

```diff
@@ -225,15 +246,48 @@ class MatchGlobalFwd
   double matchMFT_MCH_TracksAllParam(const TrackLocMCH& mchTrack, const TrackLocMFT& mftTrack);
   /// Hiroshima's Matching
   double matchHiroshima(const TrackLocMCH& mchTrack, const TrackLocMFT& mftTrack);
+  /// MC label matching
+  double noMatchFcn(const TrackLocMCH& mchTrack, const TrackLocMFT& mftTrack)
+  {
+    throw std::runtime_error("noMatchFcn function must not be called!");
+  }
 
   /// Cut functions
   bool cutDisabled(const TrackLocMCH& mchTrack, const TrackLocMFT& mftTrack) { return true; };
 
+  bool matchCut3Sigma(const TrackLocMCH& mchTrack, const TrackLocMFT& mftTrack)
+  {
+    auto dx = mchTrack.getX() - mftTrack.getX();
+    auto dy = mchTrack.getY() - mftTrack.getY();
+    auto dPhi = mchTrack.getPhi() - mftTrack.getPhi();
+    auto dTanl = TMath::Abs(mchTrack.getTanl() - mftTrack.getTanl());
+    auto dInvQPt = TMath::Abs(mchTrack.getInvQPt() - mftTrack.getInvQPt());
+    auto distance = TMath::Sqrt(dx * dx + dy * dy);
+    auto cutDistance = 3 * TMath::Sqrt(mchTrack.getSigma2X() + mchTrack.getSigma2Y());
+    auto cutPhi = 3 * TMath::Sqrt(mchTrack.getSigma2Phi() + mftTrack.getSigma2Phi());
+    auto cutTanl = 3 * TMath::Sqrt(mchTrack.getSigma2Tanl() + mftTrack.getSigma2Tanl());
+    auto cutInvQPt = 3 * TMath::Sqrt(mchTrack.getSigma2InvQPt() + mftTrack.getSigma2InvQPt());
```

Comment:
instead of calculating so many sqrt, why don't you compare squares?

Reply:
Thanks. It is done on new commit.